### PR TITLE
[Chore](statistic) do not use memory order relaxed on QueryStatistics and add sync on te…

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
1. do not use memory order relaxed on QueryStatistics
2. remove some unused code
3. add sync to test dry_run

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

File: be/src/runtime/query_statistics.h [hunk 1]
```diff
@@ -53,82 +52,44 @@ class QueryStatistics {
...
-        shuffle_send_rows.store(0, std::memory_order_relaxed);
-        _scan_bytes_from_local_storage.store(0);
-        _scan_bytes_from_remote_storage.store(0);
-
-        returned_rows.store(0, std::memory_order_relaxed);
-        max_peak_memory_bytes.store(0, std::memory_order_relaxed);
-        clearNodeStatistics();
-        //clear() is used before collection, so calling "clear" is equivalent to being collected.
-        set_collected();
+        current_used_memory_bytes += current_used_memory;
```
Comment (wangbo) [inline be/src/runtime/query_statistics.h:80]: current_used_memory_bytes should be replace

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by wangbo — APPROVED** [review]
LGTM

**Review by Mryange — APPROVED** [review]

**Pull request merged**